File: client-app/src/app/models/anecdote.ts
export interface IAnecdote {
    id: number
    title: string
    description: string
    category: string
    date: Date
    city: string
    venue: string
    isFaved: boolean
    isOwner: boolean
    favees: IFavee[]
}

export class AnecdoteFormValues implements Partial<IAnecdote> {
    id?: number = undefined
    title: string = ''
    category: string = ''
    description: string = ''
    date?: Date = undefined
    city: string = ''
    venue: string = ''

    constructor(init?: IAnecdote) {
        Object.assign(this, init)
    }
}

export interface IFavee {
    username: string
    displayName: string
    image: string
    isOwner: boolean
}

File: Domain/Anecdote.cs
using System;
using System.Collections.Generic;

namespace Domain
{
    public class Anecdote
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public string Description { get; set; }
        public string Category { get; set; }
        public DateTime Date { get; set; }
        public string City { get; set; }
        public string Venue { get; set; }
        public virtual ICollection<UserAnecdote> UserAnecdotes { get; set; }
    }
}